### DISCUSSION TITLE
How to use ezdxf to save a PNG image into a DXF file？

### DISCUSSION THREAD

**Discussion by shine-flower** [mozman/ezdxf#679] [Q&A]
After the PNG image is deleted, only the file path is left in the DXF file using the following method.
Is there a way to read PNG images into binary data and save them in DXF? I found that DXF's THUMBNAILIMAGE can add binary images, but ezdxf didn't find a way to do that.
``` text
doc.add_image_def(filename=file_name, size_in_pixel=(int(qr_code_size), int(qr_code_size)))
msp.add_image(
                insert=(float(qr_code_coordinate_x), float(qr_code_coordinate_y)),
                size_in_units=(int(qr_code_length), int(qr_code_length)),
                image_def=my_image_def,
                rotation=0
            )
```

**Comment by mozman** [comment]
The DXF format does not support storing images as binary data inside of DXF files. `Ezdxf` does not support creating thumbnail images because they are just a waste of space.

**Reply by shine-flower** [reply]
Thank you for answering this question
